### ISSUE TITLE
training problem

### ISSUE THREAD

**Issue by SimonLliu** [Sausage-SONG/Few-shot-action-recognition#3]
Hello, thanks for your coding. And i have the following uncertainty.

a) After i pretrained the I3D and TCN using MoCo, i do not see the loading interface in the **train.py** ( except args.checkpoint)

b) When i use you example_finegym.json and other default hyperparameters, i just achieve **37.3%** accuracy without MoCo pretrained. But this accuracy in the paper is **72.42%**. And the one modification is that i modified the video_folders path in **dataset.py**,
(398:video_folders = [os.path.join(data_folder+'/'+class_name, vid) for vid in video_folders]) 
because my finegym datasets has the structure of finegym_folder/class/video/frame.

Thanks!

**Comment by Sausage-SONG** [comment]
Hi Simon,

a) I loaded them by creating a new folder (args.checkpoint) and move the models there manually. Yes, this is a bit inconvenient.

b1) Regarding the dataset layout, I didn't download the finegym myself, it was already in the machine by another fellow student. So I wrote all the codes according to that layout 'finegym/video/frame'. Sorry that I didn't realize this could be a problem. Now if your dataset is 'class/video/frame', can you try to use 'StandardDataset' in dataset.py? (around [Line141](https://github.com/Sausage-SONG/Few-shot-action-recognition/blob/c07dd2a3e523a231ef8c73fb3a2b52e62b301aa0/train.py#L141) in train.py, force it to use StandardDataset) I am not sure if this can work out.

b2) Regarding the accuracy, yes, we met this problem. Our observation was that the training would be stuck when training from scratch, and the gradient is very small. We tried many ways, and we found that if we first used MSE loss to bring the model out of this trap and then use CTC loss, it would be better after that. Later we always trained models based on previous ones, so we didn't go deep into this question. I actually don't know why this would happen in theory, probably because CTC loss has some problem at the random level, or did I misuse CTC in my codes. Everything looks fine except at the beginning from scratch. Anyway, I will upload our trained models for your reference.

**Comment by SimonLliu** [comment]
> Regarding the accuracy, yes, we met this problem. Our observation was that the training would be stuck when training from scratch, and the gradient is very small. We tried many ways, and we found that if we first used MSE loss to bring the model out of this trap and then use CTC loss, it would be better after that. Later we always trained models based on previous ones, so we didn't go deep into this question. I actually don't know why this would happen in theory, probably because CTC loss has some problem at the random level, or did I misuse CTC in my codes. Everything looks fine except at the beginning from scratch. Anyway, I will upload our trained models for your reference.

Thanks. I still have another question. When training MoCo using your default file and default hyperparameters, the cost time is less than 3 minutes. Is that a common phenomenon? 
And wish to see your trained models!
Thank you!

**Comment by SimonLliu** [comment]
And i try different combination (mse+ctc, mse and ctc ) today. But i find that they all do not work and the loss's value decreases from **2.2** to **1.88**. Then the loss will not decrease and the accuracy is **33%-37%**.

**Comment by Athuliva** [comment]
is the trained model published somewhere to try it out.
